src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -191,7 +191,18 @@ private Imp makeImp(Imp imp, ExtImpRubicon rubiconImpExt) {
         if (video != null) {
             builder.video(makeVideo(video, rubiconImpExt.getVideo()));
         } else {
-            builder.banner(makeBanner(imp.getBanner()));
+            final List<Integer> sizes = rubiconImpExt.getSizes();
+            if (sizes != null) {
+                if (RubiconSize.idToSize(sizes).isEmpty()) {
+                    throw new PreBidException("Bad request.imp[].ext.rubicon.sizes");
+                } else {
+                    final List<Format> formats = RubiconSize.idToSize(sizes);
+                    final Banner modifiedBanner = Banner.builder().format(formats).build();
```

Comment:
I think we should not create new banner from scratch. Instead i propose to rework `else` banner section at all:
```
            final List<Format> overriddenSizes;
            final List<Integer> sizeIds = rubiconImpExt.getSizes();
            if (sizeIds != null) {
                final List<Format> resolvedSizes = RubiconSize.idToSize(sizeIds);
                if (resolvedSizes.isEmpty()) {
                    throw new PreBidException("Bad request.imp[].ext.rubicon.sizes");
                }
                overriddenSizes = resolvedSizes;
            } else {
                overriddenSizes = null;
            }

            builder.banner(makeBanner(imp.getBanner(), overriddenSizes));
```
then method `makeBanner(...)` can be like:
```
    private static Banner makeBanner(Banner banner, List<Format> overriddenSizes) {
        final List<Format> sizes = overriddenSizes != null ? overriddenSizes : banner.getFormat();
        return banner.toBuilder()
                .format(sizes)
                .ext(Json.mapper.valueToTree(makeBannerExt(sizes)))
                .build();
    }
```